File: euler9.py
#!/usr/bin/python

from sys import exit

def is_triplet(a, b, c):
	return (a ** 2 + b ** 2 == c ** 2)

for a in range(1, 1000):
	for b in range(a + 1, (1000 - a) / 2):
		c = 1000 - a - b
		if is_triplet(a, b, c):
			print "%d * %d * %d = %d" %(a, b, c, a * b * c)

File: euler9.c
#include <stdio.h>

int is_triplet(int a, int b, int c)
{
	return ((a * a + b * b) == (c * c));
}

int main(void)
{
	for (int a = 1; a < 1000; a++)
		for (int b = a + 1; b < (1000 - a) / 2; b++) {
			int c = 1000 - a - b;
			if (is_triplet(a, b, c)) {
				printf("%d %d %d = %d\n", a, b, c,
				       a * b * c);
				return 0;
			}
		}
}
